pcsx2/GS/Renderers/HW/GSRendererHW.cpp

```diff
@@ -2197,10 +2204,18 @@ void GSRendererHW::Draw()
 			tgt = nullptr;
 		}
 		const bool possible_shuffle = ((rt_32bit && GSLocalMemory::m_psm[m_cached_ctx.FRAME.PSM].bpp == 16) || m_cached_ctx.FRAME.Block() == m_cached_ctx.TEX0.TBP0) || IsPossibleChannelShuffle();
+		const bool req_color = m_context->ALPHA.IsUsingCs();
+		const bool req_alpha = m_context->TEX0.TCC && (m_cached_ctx.FRAME.FBMSK & (fm_mask & 0xFF000000)) != (fm_mask & 0xFF000000);
 
-		src = tex_psm.depth ? g_texture_cache->LookupDepthSource(TEX0, env.TEXA, MIP_CLAMP, tmm.coverage, possible_shuffle, m_vt.IsLinear(), m_cached_ctx.FRAME.Block()) :
-								g_texture_cache->LookupSource(TEX0, env.TEXA, MIP_CLAMP, tmm.coverage, (GSConfig.HWMipmap >= HWMipmapLevel::Basic || GSConfig.TriFilter == TriFiltering::Forced) ? &hash_lod_range : nullptr, possible_shuffle, m_vt.IsLinear(), m_cached_ctx.FRAME.Block());
+		src = tex_psm.depth ? g_texture_cache->LookupDepthSource(TEX0, env.TEXA, MIP_CLAMP, tmm.coverage, possible_shuffle, m_vt.IsLinear(), m_cached_ctx.FRAME.Block(), req_color, req_alpha) :
+								g_texture_cache->LookupSource(TEX0, env.TEXA, MIP_CLAMP, tmm.coverage, (GSConfig.HWMipmap >= HWMipmapLevel::Basic || GSConfig.TriFilter == TriFiltering::Forced) ? &hash_lod_range : nullptr,
+															possible_shuffle, m_vt.IsLinear(), m_cached_ctx.FRAME.Block(), req_color, req_alpha);
 
+		if (src && !src->IsFromTarget() && GSUtil::GetChannelMask(src->m_TEX0.PSM) == 0xf && TEX0.TBP0 != m_cached_ctx.FRAME.Block() && TEX0.TBP0 != m_cached_ctx.ZBUF.Block())
+		{
+			// Kill any possible targets we missed, they might be wrong now.
```

Comment:
Doing this when it hits the source map seems a bit questionable. Maybe it'd be better in CreateSource().

But I'm still wondering why it's a thing. Shouldn't invalidation take care of removing any targets?

Reply:
Targets don't always get completely overridden, for example it will make a 640x224 texture to do some sort of screen copy thing, then it will toss that out by writing a 128x128 texture over it. Do we keep the old target around? I don't think we should, it'll cause nothing but problems, but invalidation won't know that.